Crate lib doesn't build without features

Crate `lib` doesn't build without features. This is the output building the crate without features:
```
1/10: no features
warning: unreachable statement
  --> nano-primitives/src/pk_tree.rs:23:5
   |
20 |     return Default::default();
   |     ------------------------- any code following this expression is unreachable
...
23 |     assert_eq!(public_keys.len(), SLOTS as usize);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ unreachable statement
   |
   = note: `#[warn(unreachable_code)]` on by default
   = note: this warning originates in the macro `assert_eq` (in Nightly builds, run with -Z macro-backtrace for more info)

warning: unused variable: `public_keys`
  --> nano-primitives/src/pk_tree.rs:18:26
   |
18 | pub fn pk_tree_construct(public_keys: Vec<G2Projective>) -> Vec<u8> {
   |                          ^^^^^^^^^^^ help: if this is intentional, prefix it with an underscore: `_public_keys`
   |
   = note: `#[warn(unused_variables)]` on by default

warning: variant is never constructed: `Ban`
  --> network-libp2p/src/connection_pool/handler.rs:19:5
   |
19 |     Ban { ip: IpNetwork },
   |     ^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(dead_code)]` on by default

error[E0432]: unresolved import `parking_lot`
 --> lib/src/client.rs:3:5
  |
3 | use parking_lot::RwLock;
  |     ^^^^^^^^^^^ use of undeclared crate or module `parking_lot`

warning: unused imports: `KeyPair`, `PrivateKey`
  --> lib/src/config/config.rs:19:18
   |
19 | use nimiq_keys::{KeyPair, PrivateKey};
   |                  ^^^^^^^  ^^^^^^^^^^
   |
   = note: `#[warn(unused_imports)]` on by default

For more information about this error, try `rustc --explain E0432`.
error: could not compile `nimiq-lib` due to previous error; 1 warning emitted
```
Issue was introduced in 34f4c03833f6ca950d6c14794b961063c46feba9.

--- a/lib/Cargo.toml
+++ b/lib/Cargo.toml
@@ -30,7 +30,7 @@ hex = "0.4"
 lazy_static = "1.4"
 log = "0.4"
 log-panics = { version = "2.0", features = ["with-backtrace"], optional = true }
-parking_lot = { version = "0.11", features = ["deadlock_detection"], optional = true }
+parking_lot = { version = "0.11", features = ["deadlock_detection"] }
 paw = "1.0"
 rand = "0.7"
 serde = "1.0"
@@ -64,12 +64,12 @@ nimiq-validator-network = { path = "../validator-network", optional = true }
 nimiq-wallet = { path = "../wallet", optional = true }
 
 [features]
-deadlock = ["parking_lot"]
+deadlock = []
 default = []
 launcher = []
 logging = ["fern", "colored"]
 metrics-server = ["nimiq-metrics-server"]
 panic = ["log-panics"]
-rpc-server = ["validator", "nimiq-rpc-server", "nimiq-wallet", "parking_lot"]
+rpc-server = ["validator", "nimiq-rpc-server", "nimiq-wallet"]
 validator = ["nimiq-validator", "nimiq-validator-network", "nimiq-bls", "nimiq-rpc-server"]
 wallet = ["nimiq-wallet"]

